doc/sphinx/source/fusion/index.rst

```diff
@@ -5,38 +5,47 @@ Optimize Graphs
-
-In handling a :term:`function graph`, there are many ways to describe what 
-happens when we translate the framework's output of ops into an nGraph 
-graph. :term:`Fusion` is the term we shall use in our documentation, but the the 
-action also can be described as: *combining*, *folding*, *collapsing*, or 
-*merging* of graph functions. The most common use case is to *fuse* a subgraph 
-from the function graph into :doc:`one of the nGraph Core ops <../ops/index>`. 
+----------------------------
+
+The nGraph Compiler is an optimizing compiler. As such, it provides a way to 
+capture a given :term:`function graph` and perform a series of optimization 
+passes over that graph. The result is a semantically-equivalent graph that, when 
+executed using any |InG| :doc:`backend <../programmable/index>`, has optimizations 
+inherent at the hardware level: superior runtime characteristics to increase 
+training performance or reduce inference latency.   
+
+There are several ways to describe what happens when we capture and translate 
+the framework's output of ops into an nGraph graph. :term:`Fusion` is the term 
+we shall use in our documentation; the action also can be described as: 
+*combining*, *folding*, *squashing*, *collapsing*, or *merging* of graph 
+functions. The most common use case is to *fuse* a subgraph from the function 
+graph into :doc:`one of the nGraph Core ops <../ops/index>`.
 
 Optimization passes may include algebraic simplifications, domain-specific 
 simplifications, and fusion. Most passes share the same mode of operation (or 
-the same operational structure) and consist of two stages:
+the same operational structure) and consist of various stages (each one a 
+:term:`step`) where a developer can experiment with the intercepted or dynamic 
+graph. These steps may by cycled or recycled as needed: 
 
-#. Locating a list of potential transformation candidates (usually, subgraphs) 
-   in the given graph.
-#. Transforming the selected candidates into semantically-equivalent subgraphs 
-   that run faster and/or with less memory.
+#. Locate a list of potentially-transformable subgraphs in the given graph.
+#. Transform the selected candidates into semantically-equivalent subgraphs 
+   that execute faster, or with less memory (or both). 
+#. Verify that the optimization pass performs correctly, with any or all expected 
+   transformations, with the ``NGRAPH_SERIALIZE_TRACING`` option, which 
+   serializes a graph in the `json` format after a pass.
+#. Measure and evaluate your performance improvements with ``NGRAPH_CPU_TRACING``, 
+   which produces timelines compatible with ``chrome://tracing``.
 
-Optimization passes can be programmed ahead of time if you know what your graph 
-will look like when it's ready to be executed, or the optimization passes can 
-be figured out manually with *Interpreter* mode on a stateless graph. 
+Optimization passes can be programmed ahead of time if you know or can predict 
```

Comment:
Optimizations can be experimented without using any backend by registering a pass with pass manager(`Manager`) and calling `run_passes` on a Function and then inspecting a transformed graph.
Interpreter isn't different from any other backend, so optimization strategies are hard-coded and can't be changed dynamically.